pkg/infrastructure/openstack/postprovision/floatingips.go

```diff
@@ -40,11 +41,20 @@ func FloatingIPs(ctx context.Context, c client.Client, cluster *capo.OpenStackCl
 		return err
 	}
 
-	_, err = createAndAttachFIP(networkClient, "bootstrap", infraID, cluster.Status.ExternalNetwork.ID, bootstrapPort.ID)
+	fip, err := createAndAttachFIP(networkClient, "bootstrap", infraID, cluster.Status.ExternalNetwork.ID, bootstrapPort.ID)
 	if err != nil {
 		return err
 	}
 
+	bootstrapMachine.Status.Addresses = append(bootstrapMachine.Status.Addresses, apiv1.NodeAddress{
+		Type:    apiv1.NodeExternalIP,
+		Address: fip.FloatingIP,
+	})
+
+	if err := c.Update(ctx, bootstrapMachine); err != nil {
+		return fmt.Errorf("failed to update the bootstrap Machine manifest with the floating ip: %w", err)
+	}
```

Comment:
As discussed on Slack, this is nasty, but we'll probably get away with it.

The problem is that we're writing to status owned by another controller. It's likely to be overwritten.

The reason we'll most likely get away with it is that because we added the FIP manually, if it is overwritten the new status should also contain the same NodeExternalIP entry.

As this is a red flag, can we add a comment explaining why we're doing it and why we think we will get away with it?